src/components/dropdownSearch/tests/DropdownSearch.spec.tsx

```diff
@@ -230,14 +250,35 @@ describe('DropdownSearch', () => {
         };
       });
 
-      it('should show the filterBox if the dropdown is open', () => {
+      it('should show the filterBox if the dropdown is open and search is on', () => {
+        renderDropdownSearch(_.extend({}, ownProps, {
+          isOpened: true,
+        }));
+
+        expect(dropdownSearch.find(FilterBox).length).toBe(1);
+      });
+
+      it('should show the button if the dropdown is open and search is off and supportSingleCustomOption is true', () => {
+        const infiniteSerchThresold = 1000000000000;
```

Comment:
`const infiniteSearchThreshold = Infinity;` (use infinity and you had two typos)

Reply:
not a fan of Infinity because of it's "specialness" , such as Math.ceil(1/Infinity) === 0 to be true, which may be mathematically correct, but leaves it behaving differently than other regular numbers.

Reply:
had a specific case for the pagination calculation in TableDataModifier.ts where Infinity was causing problems and not 1000000000000